buildSrc/src/main/groovy/org/elasticsearch/gradle/vagrant/Installation.groovy

```diff
@@ -0,0 +1,47 @@
+package org.elasticsearch.gradle.vagrant
+
+/**
+ * Describes the installation of an external dependency of the build environment,
+ * like Vagrant, or VirtualBox. Provides facilities for determining if the command
+ * is installed, if it is the correct version, which version was found, and any
+ * errors that were encountered when checking for the installation.
+ */
+class Installation {
```

Comment:
I don't think we really need this abstraction...can we just keep it simple and store a boolean property on the root project indicating whether vagrant is installed at the given version, and the same for virtualbox? You can even store this in a map, along with whatever error description you want about getting the version. eg.

```
rootProject.ext.vagrant = ['supported': false, 'info': 'Unsupported version of vagrant [${version}]. Need [Vagrant 1.8.6+]']
....
if (rootProject.vagrant.supported == false) {
    logger.warn(rootProject.vagrant.info)
} else {
    // use vagrant
}
```